cmd/bad-key-revoker/main_test.go

```diff
@@ -55,24 +55,32 @@ func TestSelectUncheckedRows(t *testing.T) {
 	fc := clock.NewFake()
 
 	bkr := &badKeyRevoker{
-		dbMap:  dbMap,
-		logger: blog.NewMock(),
-		clk:    fc,
+		dbMap:             dbMap,
+		logger:            blog.NewMock(),
+		clk:               fc,
+		maxReplicationLag: time.Second * 22,
 	}
 
 	hashA, hashB, hashC := randHash(t), randHash(t), randHash(t)
+
+	// insert a blocked key that's marked as already checked
 	insertBlockedRow(t, dbMap, fc, hashA, 1, true)
 	count, err := bkr.countUncheckedKeys(ctx)
 	test.AssertNotError(t, err, "countUncheckedKeys failed")
 	test.AssertEquals(t, count, 0)
 	_, err = bkr.selectUncheckedKey(ctx)
 	test.AssertError(t, err, "selectUncheckedKey didn't fail with no rows to process")
 	test.Assert(t, db.IsNoRows(err), "returned error is not sql.ErrNoRows")
+
+	// insert a blocked key that's due to be checked
+	fc.Set(time.Now().Add(-bkr.maxReplicationLag - time.Second))
 	insertBlockedRow(t, dbMap, fc, hashB, 1, false)
+	fc.Set(time.Now())
```

Comment:
This really makes me feel like the ideal fakeClock would have a method that returns a _copy of_ the fake clock, but with the time set to some other time. That way you could inline the first of these calls, and not have to do the reset.

But that's a whole other thing that shouldn't sidetrack you :D